scripts/SubsetSelection/src/encoders/bge.py

```
@@ -0,0 +1,276 @@
+            return hidden_states[:, 0]
+        elif self.config['pooling_method'] == 'mean':
+            s = torch.sum(hidden_states * attention_mask.unsqueeze(-1).float(), dim=1)
+            d = attention_mask.sum(axis=1, keepdim=True).float()
+            return s / d
+        elif self.config['pooling_method'] == 'last_token':
+            left_padding = (attention_mask[:, -1].sum() == attention_mask.shape[0])
+            if left_padding:
+                return hidden_states[:, -1]
+            sequence_lengths = attention_mask.sum(dim=1) - 1
+            batch_size = hidden_states.shape[0]
+            return hidden_states[torch.arange(batch_size, device=hidden_states.device), sequence_lengths]
```

Pylint (3.3.8)

[refactor] Unnecessary "elif" after "return", remove the leading "el" from "elif"

(R1705)